osu.Game.Rulesets.Catch/Skinning/LegacyCatchHitObjectPiece.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using JetBrains.Annotations;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Pooling;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Rulesets.Catch.Objects.Drawables;
+using osu.Game.Rulesets.Catch.UI;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Skinning;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Rulesets.Catch.Skinning
+{
+    public abstract class LegacyCatchHitObjectPiece : PoolableDrawable
+    {
+        public readonly Bindable<Color4> AccentColour = new Bindable<Color4>();
+        public readonly Bindable<bool> HyperDash = new Bindable<bool>();
+
+        private readonly Sprite colouredSprite;
+        private readonly Sprite overlaySprite;
+        private readonly Sprite hyperSprite;
+
+        [Resolved]
+        private ISkinSource skin { get; set; }
+
+        protected ISkinSource Skin => skin;
+
+        [Resolved(canBeNull: true)]
+        private DrawableHitObject drawableHitObject { get; set; }
+
+        [CanBeNull]
+        protected DrawablePalpableCatchHitObject DrawableHitObject => (DrawablePalpableCatchHitObject)drawableHitObject;
```

Comment:
What's the reason for this null-permissiveness? Removing the `CanBeNull` annotations doesn't seem to fail any tests, and I'd surely like to think any of these pieces should really rely on a DHO existing above them (osu! DHO pieces make these sorts of assumptions already).

Reply:
The nullability is not necessary for now.
The plain is don't use DHO for objects on the plate (see `Catcher.createCaughtObject`).
And the piece classes will be used in both DHOs and caught objects like `CaughtFruit { FruitPiece }` and `DrawableFruit { FruitPiece }`.
For a caught object, this `DrawableHitObject` will be null.